explorer/explorer.go

```diff
@@ -148,12 +149,70 @@ type explorerUI struct {
 	ChainParams     *chaincfg.Params
 	Version         string
 	NetName         string
+	// DisplaySyncStatusPage indicates if the sync status page is the only web
+	// page that should be accessible during DB synchronization.
+	DisplaySyncStatusPage bool
 }
 
 func (exp *explorerUI) reloadTemplates() error {
 	return exp.templates.reloadTemplates()
 }
 
+// SyncStatusUpdate receives the raw progress updates calculates the percentage
+// and updates the blockchainSyncStatus.ProgressBars.
+func (exp *explorerUI) SyncStatusUpdate(barLoad chan *dbtypes.ProgressBarLoad) {
+	// when this method is run check if it should be running in the first place.
+	exp.NewBlockDataMtx.RLock()
+	if exp.DisplaySyncStatusPage {
+		return
```

Comment:
`exp.NewBlockDataMtx` is still locked when you return.  Need to unlock it before return.